bar.go

```diff
@@ -89,7 +89,8 @@ func newBar(desk Desktop) fyne.CanvasObject {
 	if desk.WindowManager() != nil {
 		desk.WindowManager().AddStackListener(appBar)
 	}
-	for _, app := range appBar.desk.IconProvider().DefaultApps() {
+	for _, name := range Instance().Settings().DefaultApps() {
```

Comment:
I realise in the morning light that there are too many Instance() lookups - we should try and avoid them.
Right above us we are passed the desktop instance as a variable so that should be used.